boa: add generator support (#614)
alibaba/pipcook: docs/manual/intro-to-boa.md

// "b'foobar'"
```

> ### Use of Generator

Reviewer: This should be prepended before https://github.com/alibaba/pipcook/blob/49b821201e61fb17dc63cef31806bec675b531c5/docs/manual/intro-to-boa.md#python-functions-in-worker_threads.

> `Boa` will also handle `Generator` class in the native python runtime. JavaScript syntax of `Generator` could be applied to Python `Generator` directly:

Reviewer: ```suggestion
The package is able to handle the Python generator class in JavaScript directly:
```

```js

> 

Reviewer: Remove this line.

> const generator = generatorFromPython(); // Generator

Reviewer: I think it's better to represent it from Python:

- prepend a python block to write a generator function.
- import the generator object using boa.import here.

$ node --experimental-modules --experimental-loader @pipcook/boa/esm/loader.mjs app.mjs
```

> ### Python Generators in JS

Reviewer: ```suggestion
### Generators
```

> The package is able to handle the Python generator class in JavaScript directly:

Reviewer: ```suggestion
The package is able to handle the Python generators in JavaScript directly:
```

```python
# Write a Python Generator in count_down.py
def count_down(count):
  while count >= 0:
    yield count
    count -= 1
```

> 

Reviewer: We need to describe the following code block.

```js
const boa = require('@pipcook/boa')

> const countDown = boa.import('path_to_above_py')

Reviewer: ```suggestion
const countDown = boa.import('count_down')
```

const generator = countDown(3); // Generator

// You can access data via the following syntax
for (const element of generator) {
  console.log(element) // 3 2 1 0
}

// or use typical next syntax

> const generator = countDown(3); // Generator

Reviewer: I think this example should be the first case, and it's good to split into different blocks.
